pkg/chains/objects/objects.go

```diff
@@ -0,0 +1,196 @@
+package objects
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"github.com/tektoncd/pipeline/pkg/client/clientset/versioned"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	pipelineRunKind = "PipelineRun"
+	taskRunKind     = "TaskRun"
+)
+
+type Annotation struct {
+	Err   error
+	Value string
+	Ok    bool
+}
+
+type Result struct {
+	Name  string
+	Value string
+}
+
+// Represents a generic K8s object
+// This isn't meant to be a final implementation, just one approach
+// Many of these methods can be abstracted away further
+type K8sObject interface {
```

Comment:
You might be interested in some prior art in https://github.com/tektoncd/results/tree/main/pkg/watcher/reconciler - which watched both TaskRuns and PipelineRuns to update annotations for Results :)

The premise is roughly the same, but the other way makes heavy use of unstructured.Unstructured + defining the client to use when creating the reconciler.

Reply:
Okay i see you already have an implementation of a generic object there, i think we'll just re-use that implementation - what do you think?